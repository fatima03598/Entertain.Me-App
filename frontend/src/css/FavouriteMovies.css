.favouritesContainer{
    display:flex;
    flex-direction: row;
    width:100vw;
}
h1{
    width:100%;
}
.moodSection {
   
    width:33.3%;
    border: 2px solid black;
    display:flex;
    flex-direction:row;
    flex-wrap:wrap;
    padding: 70px;
}

.moodSectionMovies{
    display:flex;
    flex-direction:row;
    height:50vh;
    width:100%;
    align-items:center;
}

.mood-section {
    background-color: #4C4C58;
}