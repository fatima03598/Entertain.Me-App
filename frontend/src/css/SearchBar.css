.searchContainer{
    display:flex;
    flex-direction:row;
    width:50%;
    justify-self:center;
    border:2px black solid;
}
.searchBox{
    border: black 2px solid;
    width:30%;
    display: flex;
    flex-direction: row;
}

#searchBar{
    background-color:red; 
}