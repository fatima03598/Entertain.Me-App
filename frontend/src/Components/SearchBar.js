import React from 'react';
import Form from "react-bootstrap/Form";
import Button from 'react-bootstrap/Button';
import {Navbar, Nav} from 'react-bootstrap';
import '../css/SearchBar.css';
import Movie from '../Components/Movie';
import { Link, Redirect } from "react-router-dom";

class SearchBar extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            movieData:null,
            searchQuery:null,
            pageNumber:1,
            pageSection:0
        }
    }

    handleClickMovieSearch = (e) => {

        let key = "ca3b3298e0c4d85c79e20c33b747a10c"

    
        console.log("Clicked")
        console.log(this.state.searchQuery)
        let search = this.state.searchQuery

        fetch(`https://api.themoviedb.org/3/search/movie?api_key=${key}&language=en-US&query=${search}&page=1&include_adult=false`)
        .then(response => response.json())
        .then(data =>{
            console.log(data.results) //testing
            this.setState({
                movieData : data.results,
                pageSection : 1
            })
            this.props.movieDataExists();
        
        })
        .catch(error => {
            console.log(error)
        })

        // this.props.movieDataExists();
        // this.props.forceRender();

    }

    handleClickNext = (e) => {
        if(this.state.pageSection == 1){
            this.setState({pageNumber : this.state.pageNumber + 1})
        }
        
        let key = "ca3b3298e0c4d85c79e20c33b747a10c"
        let search = this.state.searchQuery

        console.log(`https://api.themoviedb.org/3/search/movie?api_key=${key}&language=en-US&query=${search}&page=${this.state.pageNumber}&include_adult=false`)
    
        fetch(`https://api.themoviedb.org/3/search/movie?api_key=${key}&language=en-US&query=${search}&page=${this.state.pageNumber}&include_adult=false`)
        .then(response => response.json())
        .then(data =>{
            console.log(data.results) //testing
            if(this.state.pageSection == 0){
                this.setState({
                    movieData : data.results,
                    pageSection : 1
                })

            }
            else{
                this.setState({
                    movieData : data.results,
                    pageSection : 0
                })
            }
        })
        .catch(error => {
            console.log(error)
        })
    }

    handleClickMovieGet = (e) => {

        console.log("Get movie clicked")

        fetch('/movie/1')
        .then(response =>{
            console.log(response)
        })
        .catch(error =>{
            console.log(error)
        })
    }

    handleSearchChange = (e) => {
        let searchQuery = e.target.value;
        this.setState({searchQuery : searchQuery})
    }

    getUserFavourites = () => {

        fetch(`/favourites/${this.props.userId}`)
        .then(response => response.json())
        .then(data => {
            console.log(data)
        })
        .catch(error =>{
            console.log(error)
        })
    }

    login = (event) => {
        fetch('/login')
    }

    handleClickGoBack = () => {
        console.log("go back clicked")
        this.props.goBack();
        this.setState({movieData:null});
    }


    render() {

        if(this.state.movieData){
            if(this.state.pageSection == 1){
                while((this.state.movieData).length > 10){
                    this.state.movieData.pop();
                }
            }else{
                while((this.state.movieData).length > 10){
                    this.state.movieData.shift();
                }
            }
        }

            return(
                <div>
                    <div className="searchContainer">
                        <div className="searchBox">
                            <Form>
                                <input id="searchBar" onChange={this.handleSearchChange} type="text" placeholder="Search a movie"></input>
                                <Button onClick={this.handleClickMovieSearch}>üîç</Button>
                            </Form>
                        </div>

                        {this.state.movieData ? <div className="results-btn-container"><Button className="results-button" onClick={this.handleClickGoBack}>Favourites</Button><Button className="btn btn-success" onClick={this.handleClickNext}>Next Page</Button> </div>: null}
                        
                    </div>

                    {this.state.movieData ? <div className="moviesContainer">
                        {this.state.movieData.map(data => <Movie userId = {this.props.userId} data = {data}/>)}
                    </div> : null}
                    
                </div>
            )
        }
            
    
}

export default SearchBar;