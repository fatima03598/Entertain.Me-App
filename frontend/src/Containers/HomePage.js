import React from 'react';
// import 'bootstrap/dist/css/bootstrap.min.css';
import Form from "react-bootstrap/Form";
import Button from 'react-bootstrap/Button';
import {Navbar, Nav} from 'react-bootstrap';
import '../css/Homepage.css';
import Movie from '../Components/Movie';
import SearchBar from '../Components/SearchBar'
import { Link, Redirect } from "react-router-dom";
import FavouriteMovies from '../Components/FavouriteMovies';

class Homepage extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            userID:1,
            movieData:false,
            searchQuery:null,
            pageNumber:1,
            pageSection:0,
            force : false,
        }
    }

    getUserFavourites = () => {

        this.setState({userID : 1})

        fetch(`/favourites/${this.state.userID}`)
        .then(response => response.json())
        .then(data => {
            console.log(data)
        })
        .catch(error =>{
            console.log(error)
        })
    }

    forceRender = () => {
        if(this.state.force === false){
            this.setState({force : true})
        }
        else{
            this.setState({force : false})
        }
    }

    movieDataExists = () => {
        this.setState({movieData : true})
    }



   
    render(){
  
            // if(this.state.movieData){
            //     console.log(this.state.movieData)
            //     console.log("MOVIE DATA")
            //     return(
            //         <SearchBar forceRender={this.forceRender} movieDataExists={this.movieDataExists}/>
            //     )
            // }
            // else {
            //     console.log(this.state.movieData)
            //     console.log(" NO MOVIE DATA")
            //     return (
            //         <div>
            //             <SearchBar forceRender={this.forceRender} movieDataExists={this.movieDataExists}/>
            //             <FavouriteMovies forceRender={this.forceRender}/>
            //         </div>
            //     )
            // }

            return (
                <div>
                    <SearchBar />
                    <FavouriteMovies />
                </div>
                    
            )

            

          
        
    }
}

export default Homepage;

