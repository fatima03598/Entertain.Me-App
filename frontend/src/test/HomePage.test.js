import React from 'react';
import ReactDOM from 'react-dom'
import {shallow, mount } from 'enzyme';
import HomePage from '../Containers/HomePage';
import SearchBar from '../Components/SearchBar';
import FavouriteMovies from '../Components/FavouriteMovies';
import renderer from 'react-test-renderer';
import { findByTestAttr } from '../test/testUtils';


describe('<HomePage />', () => {
    
    const essentialProps = {
        userID: null,
        movieData:false,
    }

    let componentWithProps = <HomePage {...essentialProps}/>
    let wrapper;

    beforeEach(() => wrapper = shallow(componentWithProps));

    it('should mount to DOM', () => {
        const div = document.createElement('div');
        ReactDOM.render(componentWithProps, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(componentWithProps)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render a single <div /> if state movieData', () => {
        wrapper.setState({movieData: true})
        expect(wrapper.find('div').length).toEqual(1);
    });

    it('should render <FavouriteMovies /> if props userId not null', () => {
        
    });
    


})