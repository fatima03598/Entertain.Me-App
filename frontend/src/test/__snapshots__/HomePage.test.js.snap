// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders correctly 1`] = `
<div
  className="d-flex flex-column h-70 align-items-center justify-content-center"
>
  <div>
    <div
      className="searchContainer"
    >
      <div
        className="searchBox"
      >
        <form
          className=""
        >
          <input
            id="searchBar"
            onChange={[Function]}
            placeholder="Search a movie"
            type="text"
          />
        </form>
      </div>
    </div>
  </div>
  <div
    className="h-100 w-100 d-flex flex-column bg-light align-items-center justify-content-between "
  >
    <p
      className="mb-0 display-4"
    >
      Search for your favourites films
    </p>
    <div
      className="container w-50 d-flex flex-row align-items-center justify-content-around "
    >
      <img
        src="https://www.bestmovieposters.co.uk/wp-content/uploads/2019/01/4A8ltk8G.jpeg"
        style={
          Object {
            "height": "200px",
            "width": "133px",
          }
        }
      />
      <img
        src="https://www.mauvais-genres.com/26247/avengers-endgame-original-movie-poster-15x21-in-2019-anthony-russo-robert-downey-jr.jpg"
        style={
          Object {
            "height": "200px",
            "width": "133px",
          }
        }
      />
      <img
        src="https://images-na.ssl-images-amazon.com/images/I/41a0nad8wPL._AC_.jpg"
        style={
          Object {
            "height": "200px",
            "width": "133px",
          }
        }
      />
    </div>
    <p
      className="display-3"
    >
      â†“
    </p>
    <p
      className="display-4"
    >
      Add them to mood-specific lists
    </p>
    <div
      className="container w-50 d-flex flex-row align-items-center justify-content-around"
    >
      <p
        className="display-1"
      >
        ðŸ˜€
      </p>
      <p
        className="display-1"
      >
        ðŸ˜¥
      </p>
      <p
        className="display-1"
      >
        ðŸ¥±
      </p>
    </div>
    <p
      className="display-3"
    >
      â†“
    </p>
    <p
      className="display-4"
    >
      Get mood-matched personalised recommendations!
    </p>
  </div>
</div>
`;
